### PR TITLE
Stovol

### PR DESCRIPTION
**Pull request by phenaff**
Various small fixes:
- Consolidated stochastic volatility examples in one file
- Some progress in figuring out the crash in make_zero_coupon example for observation date 2007-04-27: the exception is now caught, but the numerical issue is still there. Probably some inconsistent day count.
- We need to upgrade to cython 0.23 in order to handle exceptions in properties. cython 0.20 seems to generate bad code.

### CODE REVIEW

**Comment by phenaff** [comment]
@dpinte: I have a failure in Python 3 due to a bad string comparison in test_term_structure. I cannot figure out how to do a string comparison in Python 3 (not installed on my machine right now).

**Comment by dpinte** [comment]
@phenaff I'll have a look asap!

**Comment by phenaff** [comment]
I think the problem is on line #226 of yield_term_structure.pyx:

DayCounter.from_name(dc.name())

**Comment by dpinte** [comment]
@phenaff  Can you try removing the `b` in `self.assertEqual(ts.day_counter.name(), b"Actual/Actual (ISDA)")` ? The internal names are using the native Python string format which is unicode in Python 3! 

We should add some explicit type checks in the `DayCounter.from_name` method to prevent such errors.

**Comment by phenaff** [comment]
@dpinte Sure, will try that today. I got a python 3.4 virtual env set up, so should be able to test under both versions.

**Comment by phenaff** [comment]
I think the issue may be in _get_daycounter_type_from_name(str name). In the following minimal example:

`from quantlib.termstructures.yields.api import FlatForward
from quantlib.time.api import Date, DayCounter, ActualActual, ISDA
day_counter = ActualActual(ISDA)
settlement_date = Date(1, 1, 2001)
risk_free_rate = 0.04
ts = FlatForward(
            reference_date=settlement_date,
            forward=risk_free_rate,
            daycounter=day_counter)

dc_1_name = day_counter.name()
dc_2 = ts.day_counter
ValueError: Unknown day counter type: b'Actual/Actual
`

The error message seems to indicate that the after parsing, "name" is of type bytes.  Works fine in 2.7.

**Comment by phenaff** [comment]
The cython [doc](http://docs.cython.org/src/tutorial/strings.html) seems to suggest to keep all strings as unicode.

**Comment by phenaff** [comment]
Not sure if this helps, but I created a minimal example as a IPython notebook, and the error message is slightly different: no 'byte' tag on string 'name'
[cython-test.pdf](https://github.com/enthought/pyql/files/134013/cython-test.pdf)

File: quantlib/util/compat.pyx [hunk 1]
```diff
@@ -29,3 +29,17 @@ cdef py_string_from_utf8_array(const char* char_array):
...
+cdef unicode _ustring(s):
+    if type(s) is unicode:
+        # fast path for most common case(s)
+        return <unicode>s
+    elif PY_MAJOR_VERSION < 3 and isinstance(s, str):
+        return s.decode('UTF-8')
+    elif PY_MAJOR_VERSION >= 3 and isinstance(s, bytes):
+        return s.decode('UTF-8')
+    elif isinstance(s, unicode):
+        return unicode(s)
```
Comment (dpinte) [inline quantlib/util/compat.pyx]: why not returning just `s`?

**Comment by dpinte** [comment]
@phenaff one question, otherwise, LGTM

**Comment by dpinte** [comment]
@phenaff Can you merge master in this branch?

**Comment by phenaff** [comment]
@dpinte Sure, will do that in the next few days

**Comment by phenaff** [comment]
@dpinte Merge completed

**Pull request merged**